### PR TITLE
Add correct extension mode to plugin context

### PR DESCRIPTION
**Pull request by eneufeld**
The plugin context was hardcoded to have the production extensionMode.
Now a flag is passed to mark a plugin as under development when
deploying it. This flag is stored in the plugin metadata and later set in
the plugin. When creating the context this flag is then converted to the
correct extensionMode.

Fix #10201

Contributed on behalf of STMicroelectronics
Signed-off-by: Eugen Neufeld <eneufeld@eclipsesource.com>

### CODE REVIEW

File: packages/plugin-ext/src/plugin/plugin-manager.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ import {
 } from '../common/plugin-api-rpc';
 import { PluginMetadata, PluginJsonValidationContribution } from '../common/plugin-protocol';
 import * as theia from '@theia/plugin';
+import * as types from './types-impl';
```
Comment (lucas-koehler) [inline packages/plugin-ext/src/plugin/plugin-manager.ts:33]: Why import all and not only `ExtensionMode`?

**Comment by lucas-koehler** [comment]
Generally looks good to me and works as expected. You probably should add an entry to the `CHANGELOG.md` though.
Further question: Did you consider that there are three extension modes? This does not consider the third one "Test" with value `3`.

File: packages/plugin-ext/src/plugin/plugin-manager.ts (hunk 1 above)
Comment (eneufeld) [reply packages/plugin-ext/src/plugin/plugin-manager.ts:33]: the export is weird, no way to import a single type

**Comment by lucas-koehler** [comment]
> > Generally looks good to me and works as expected. You probably should add an entry to the `CHANGELOG.md` though. Further question: Did you consider that there are three extension modes? This does not consider the third one "Test" with value `3`.
> 
> I will update the ChangeLOg, the test option is explicitly excluded in the ticket

Alright, I'm just wondering whether the api should support a type that makes it easier to pass the third value later without needing to change the type from boolean again. But I'm also fine with leaving it as is for now.

**Pull request closed without merging**